### PR TITLE
Work around union/polysolid boundaries and fix transformation of explicit daughters

### PR DESCRIPTION
**Pull request by sethrj**
This adds a minimal reproducer demonstrating #1260 (which is still disabled), adds a minimal GDML failure (union volume with a daughter), and then:
- Forces all LVs with a union/polysolid boundary to be explicitly inlined into the parent volume
- Fixes an incorrect doubly applied transformation to "explicit" background cells (i.e. parents with children subtracted).

The force-inlining may be very inefficient but at least allows us a working solution while we implement a better longer-term alternative. It also creates universes with *many* more surfaces than before, which means that the surface hash acceleration from #1183 needs to be merged for performance reasons for nontrivial problems.

Tilecal top-level universe:
tiletb csg

Raytrace at y=0:
tilecal-xz

Raytrace at x=1.8m still shows some issues with the GenTrap's positioning somehow:
emec-yz-x1800mm

### CODE REVIEW

File: src/orange/orangeinp/CsgTreeUtils.cc [hunk 1]
```diff
@@ -90,20 +91,26 @@ void replace_and_simplify(CsgTree* tree, NodeId repl_key, Node repl_value)
...
     // Replace nonliterals
     for (auto n : range(CsgTree::false_node_id() + 1, NodeId{tree->size()}))
     {
+        auto repl_state = state[n.get()];
         if (std::holds_alternative<Surface>((*tree)[n]))
         {
-            continue;
+            if (repl_state == NodeReplacer::unknown)
+            {
+                // Track boundary surfaces that we can't prove, likely because
```
Comment (elliottbiondo) [inline src/orange/orangeinp/CsgTreeUtils.cc]: Keep track of?

File: src/orange/orangeinp/CsgTreeUtils.cc [hunk 2]
```diff
@@ -39,7 +39,8 @@ namespace orangeinp
  *
  * \return Node ID of the lowest node that required simplification.
```
Comment (elliottbiondo) [inline src/orange/orangeinp/CsgTreeUtils.cc]: This comment needs to be updated

File: test/orange/OrangeGeant.test.cc [hunk 3]
```diff
@@ -198,6 +198,39 @@ TEST_F(SimpleCmsGeantTest, trace)
...
 
+//---------------------------------------------------------------------------//
+class TilecalPlugTest : public GeantOrangeTest
+{
+    std::string geometry_basename() const final { return "tilecal-plug"; }
+};
+
+TEST_F(TilecalPlugTest, trace)
+{
+    if (0)
```
Comment (elliottbiondo) [inline test/orange/OrangeGeant.test.cc]: Maybe add a TODO here to explain why this block is `if`-ed out, on the off chance that it is forgotten about in the future.

**Review by elliottbiondo — APPROVED** [review]

File: test/orange/OrangeGeant.test.cc (hunk 3 above)
Comment (sethrj) [reply test/orange/OrangeGeant.test.cc]: Actually the test passes now. Good catch!

**Pull request merged**